e2e/perf/get-pr-info.py

```
@@ -0,0 +1,179 @@
+
+
+def find_prs_by_head_branch(head_branch: str, head_owner: str | None = None) -> List[int]:
+    """Find open PRs for a head branch, optionally scoped to a fork/owner."""
+    if not head_branch:
+        return []
+
+    head_ref = f"{head_owner}:{head_branch}" if head_owner else head_branch
+    print(f"Searching for PRs by head: {head_ref}", file=sys.stderr)
+
+    try:
+        result = subprocess.run(
```

Ruff (0.16.2)

[error] `subprocess` call: check for execution of untrusted input

(S603)